packages/test-utils/src/reporter.ts

```diff
@@ -277,8 +277,8 @@ class ReplayReporter<TRecordingMetadata extends UnstructuredMetadata = Unstructu
       process.env.RECORD_REPLAY_METADATA_TEST_RUN_TITLE ||
       config.runTitle;
 
-    this.apiKey = process.env.REPLAY_API_KEY || process.env.RECORD_REPLAY_API_KEY || config.apiKey;
-    this.upload = !!process.env.REPLAY_UPLOAD || !!config.upload;
+    this.apiKey = config.apiKey || process.env.REPLAY_API_KEY || process.env.RECORD_REPLAY_API_KEY;
+    this.upload = "upload" in config ? !!config.upload : !!process.env.REPLAY_UPLOAD;
```

Comment:
I think it always makes sense to prefer `config` options over env variables as they are more specific. This is a rule of thumb that I have always been using and it has never failed me so far ;p